ocaml/typing/typedecl.ml

```diff
@@ -1132,7 +1138,7 @@ let check_kind_coherence env loc dpath decl =
           if not @@ Ctype.arity_matches_decl env decl (AppArgs.length args)
           then Some Includecore.Arity
           else begin
-            match Ctype.equal env false (AppArgs.to_list args) decl.type_params with
+            match Ctype.equal env false (AppArgs.to_list args) (Ctype.app_params_of_decl decl) with
```

Comment:
I think you're rewriting this check in another PR anyway, but I think this part should be directly using `type_params` because we actually want to enforce a syntactic restriction on the manifests that they are written with the declaration's parameters as arguments.